packages/angular_devkit/build_angular/src/differential-loading/utils.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as browserslist from 'browserslist';
+import * as caniuse from 'caniuse-api';
+import * as path from 'path';
+import * as ts from 'typescript';
+
+
+export function isDifferentialLoadingNeeded(
+  tsconfigPath: string, browsersListConfigFile: string | undefined = undefined): boolean {
+
+  const browserslistPath = path.dirname(tsconfigPath);
+  const es5SupportNeeded = isEs5BrowserSupportNeeded(browserslistPath, browsersListConfigFile);
+  const target = getTargetVersion(tsconfigPath);
+
+  return es5SupportNeeded && target !== ts.ScriptTarget.ES5;
+}
+
+export function getTargetVersion(tsConfigFileName: string): ts.ScriptTarget {
+
+  const config = ts.getParsedCommandLineOfConfigFile(
+    tsConfigFileName,
+    {},
+    { ...ts.sys, onUnRecoverableConfigFileDiagnostic: () => {} },
+  );
+
+  if (typeof config === 'undefined') {
+    throw new Error('could not read tsconfig: ' + tsConfigFileName);
+  }
+
+  const compilerOptions = config.options;
+  const target = compilerOptions.target;
+
+  if (typeof target === 'undefined') {
+
+    // fall back to ES5
+    return ts.ScriptTarget.ES5;
+  }
+
+  return target;
+}
+
+export function isEs5BrowserSupportNeeded(
+  browserslistPath: string, browserslistConfigFile: string | undefined = undefined): boolean {
```

Comment:
How would we know the `browserslistConfigFile` from the CLI? this can be anything which browserlist supports. While at the moment in the CLI we generate an emotionless `browserlist`, the users are free to change this as long as browserlist can read it? 

Hence, why not simply provide the workspacePath and let `browserslist` resolve it''s configuration file? Similar to how it's currently done by autoprefixer in the style config.

`browserslist` configuration can be in various files such as;

1. package.json
2. .browserslistrc
3. browserslist

Reply:
This should be the default, I agree. The current design doc also specifies that we should be able to define one using a parameter (command line parameter when calling ng build or a parameter within angular.json). I think, this would come in-handy b/c so one could build versions for different targets.

Reply:
At the moment, having different multiple browserlist configs is not supported by autoprefixer. it only supports either resolving the browserlist file as per it's internal resolution strategy or an array of browsers to support.

My point is that if we are going to provide a way for users to change the browserlist with a configuration parameter.  We need to change the logic in a couple of builders as this would also determine which prefixes would need to be used during server builder.

IMHO, if such approach will need to happen it shouldn't be the responsibility of the differential loader utils and it should be somewhere more centralised:

In general I'd like to avoid having to parse the same file multiple times.
-> parse browser list config as required
--> pass this down the differential loader
--> pass this down to the webpack styles configuration.

That said, I am not seeing what would be the users case of building the application several times using different browser versions. (I will check the design doc)

Reply:
Ok, so we leave this away for now, right?

Reply:
I removed this param now